src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  AfterContentInit,
+  Attribute,
+  Component,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  HostBinding,
+  HostListener,
+  Input,
+  OnInit,
+  Optional,
+  Output,
+  QueryList,
+  ViewEncapsulation,
+  ViewChild,
+  ViewChildren,
+  forwardRef
+} from 'angular2/core';
+
+import {ObservableWrapper, EventEmitter} from 'angular2/src/facade/async';
+import {iterateListLike} from 'angular2/src/facade/collection';
+import {Event} from 'angular2/src/facade/browser';
+
+import {MdRadioDispatcher} from './radio_dispatcher';
+
+// TODO(mtlin):
+// Ink ripple is currently placeholder.
+// Determine motion spec for button transitions.
+// Design review.
+// RTL
+// Support forms API.
+
+var _uniqueIdCounter: number = 0;
+
+@Directive({
+  selector: 'md-radio-group',
+  host: {
+    'role': 'radiogroup',
+  },
```

Comment:
I forgot to mention this to you and @traviskaufman earlier, and it can be a TODO for a follow-up PR, but we should try to use the configuration
 ```ts
changeDetection: ChangeDetectionStrategy.OnPush,
```

This tells Angular to only perform change detection when one of the `@Input` properties is changed or when there is a local event. For checkbox and radio button it _should_ "just work" since all of the bound values are primitives.